internet.go

```diff
@@ -0,0 +1,300 @@
+		(*byte)(unsafe.Pointer(&statusCode)),
+		&bufferLength,
+		&index)
+	if err != nil {
+		return 0, NewWindowsError("HttpQueryInfo", err)
+	}
+	return HTTPStatusCode(statusCode), nil
+}
+
+func (self *HTTPRequest) Send(headers string, optional []byte) error {
+	var headersRaw *uint16
+	if headers != "" {
+		headersRaw = syscall.StringToUTF16Ptr(headers)
+	}
+	var optionalRaw *byte
+	var optionalLength uint32
+	if optional != nil {
+		optionalRaw = &optional[0]
+		optionalLength = uint32(len(optional))
+	}
+	err := wrappers.HttpSendRequest(
+		self.handle,
+		headersRaw,
+		uint32(len(headers)),
+		optionalRaw,
+		optionalLength)
+	if err != nil {
+		return NewWindowsError("HttpSendRequest", err)
+	}
+	return nil
+}
+
+type InternetConnection struct {
+	InternetObject
+}
+
+func (self *InternetConnection) OpenHTTPRequest(verb string, objectName string, version string, referer string, acceptTypes []string, flags HTTPRequestFlags) (*HTTPRequest, error) {
+	var versionRaw *uint16
+	if version != "" {
+		versionRaw = syscall.StringToUTF16Ptr(version)
+	}
+	var refererRaw *uint16
+	if referer != "" {
+		refererRaw = syscall.StringToUTF16Ptr(referer)
+	}
+	var acceptTypesRaw **uint16
+	if acceptTypes != nil {
+		acceptTypesPtrs := make([]*uint16, len(acceptTypes))
+		for i := range acceptTypes {
+			acceptTypesPtrs[i] = syscall.StringToUTF16Ptr(acceptTypes[i])
+		}
+	}
+	handle, err := wrappers.HttpOpenRequest(
+		self.handle,
+		syscall.StringToUTF16Ptr(verb),
+		syscall.StringToUTF16Ptr(objectName),
+		versionRaw,
+		refererRaw,
+		acceptTypesRaw,
+		uint32(flags),
+		0)
+	if err != nil {
+		return nil, NewWindowsError("HttpOpenRequest", err)
+	}
+	return &HTTPRequest{InternetFile{InternetObject{handle: handle}}}, nil
+}
+
+type InternetSession struct {
+	InternetObject
+}
+
+func OpenInternetSession(agent string, openType InternetOpenType, flags InternetOpenFlags) (*InternetSession, error) {
+	handle, err := wrappers.InternetOpen(
+		syscall.StringToUTF16Ptr(agent),
+		uint32(openType),
+		nil,
+		nil,
+		uint32(flags))
+	if err != nil {
+		return nil, NewWindowsError("InternetOpen", err)
+	}
+	return &InternetSession{InternetObject{handle: handle}}, nil
+}
+
+func (self *InternetSession) Connect(serverName string, serverPort uint, service InternetService) (*InternetConnection, error) {
+	handle, err := wrappers.InternetConnect(
+		self.handle,
+		syscall.StringToUTF16Ptr(serverName),
+		uint16(serverPort),
+		nil,
+		nil,
+		uint32(service),
+		0,
+		0)
+	if err != nil {
+		return nil, NewWindowsError("InternetConnect", err)
+	}
+	return &InternetConnection{InternetObject{handle: handle}}, nil
+}
+
+func (self *InternetSession) OpenURL(url string, headers string, flags URLRequestFlags) (*InternetFile, error) {
+	var headersRaw *uint16
+	if headers != "" {
+		headersRaw = syscall.StringToUTF16Ptr(headers)
+	}
+	handle, err := wrappers.InternetOpenUrl(
+		self.handle,
+		syscall.StringToUTF16Ptr(url),
+		headersRaw,
+		uint32(len(headers)),
```

Comment:
No, this is right.  We want the length of the string in characters, not the size of the pointer to it.